transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -60,6 +61,18 @@
     private static final int MIN_PREMATURE_SELECTOR_RETURNS = 3;
     private static final int SELECTOR_AUTO_REBUILD_THRESHOLD;
 
+    private final IntSupplier SELECT_NOW_SUPPLIER = new IntSupplier() {
+        @Override
+        public int get() {
+            try {
+                return selectNow();
+            } catch (IOException e) {
+                logger.warn("Caught IOException during selectNow(), moving on.", e);
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)